R/util.R

```diff
@@ -83,6 +83,41 @@ util.duplicated_all <- function(x) {
   return (duplicated_downward | duplicated_upward)
 }
 
+util.na_omit <- function(x){
+  # returns all values of vector, matrix or data.frame x that are not NA,
+  # but maintains the type of x (in contrast with stats::na.omit which adds a
+  # bunch of unsolicited attributes to vectors)
+  if(is.atomic(x) & is.vector(x)){
+    return(x[!is.na(x)])
+  }
+  if(is.matrix(x)){
+    return(na.omit(x))
+  }
+  if(is.data.frame(x)){
+    return(x[complete.cases(x), ])
+  } else{
+    stop("util.na_omit is for vectors, matrices and data.frames only.")
+  }
```

Comment:
nit: you aren't using `else` consistently. In the first two cases, you're taking advantage of the fact that `return()` prevents further code execution. In this last case, you're writing `else` _even though_ it's redundant with the `return()`. I suggest you drop the last else:

```
if (is.data.frame(x)) {
  return(x[complete.cases(x), ])
}
stop("...")
```